src/test/storage/dictionary_segment_test.cpp

```diff
@@ -8,50 +9,128 @@
 #include "../../lib/storage/dictionary_segment.hpp"
 #include "../../lib/storage/value_segment.hpp"
 
-// class StorageDictionarySegmentTest : public ::testing::Test {
-//  protected:
-//   std::shared_ptr<opossum::ValueSegment<int>> vc_int = std::make_shared<opossum::ValueSegment<int>>();
-//   std::shared_ptr<opossum::ValueSegment<std::string>> vc_str = std::make_shared<opossum::ValueSegment<std::string>>();
-// };
-
-// TEST_F(StorageDictionarySegmentTest, CompressSegmentString) {
-//   vc_str->append("Bill");
-//   vc_str->append("Steve");
-//   vc_str->append("Alexander");
-//   vc_str->append("Steve");
-//   vc_str->append("Hasso");
-//   vc_str->append("Bill");
-
-//   auto col = opossum::make_shared_by_data_type<opossum::BaseSegment, opossum::DictionarySegment>("string", vc_str);
-//   auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<std::string>>(col);
-
-//   // Test attribute_vector size
-//   EXPECT_EQ(dict_col->size(), 6u);
-
-//   // Test dictionary size (uniqueness)
-//   EXPECT_EQ(dict_col->unique_values_count(), 4u);
-
-//   // Test sorting
-//   auto dict = dict_col->dictionary();
-//   EXPECT_EQ((*dict)[0], "Alexander");
-//   EXPECT_EQ((*dict)[1], "Bill");
-//   EXPECT_EQ((*dict)[2], "Hasso");
-//   EXPECT_EQ((*dict)[3], "Steve");
-// }
-
-// TEST_F(StorageDictionarySegmentTest, LowerUpperBound) {
-//   for (int i = 0; i <= 10; i += 2) vc_int->append(i);
-//   auto col = opossum::make_shared_by_data_type<opossum::BaseSegment, opossum::DictionarySegment>("int", vc_int);
-//   auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int>>(col);
-
-//   EXPECT_EQ(dict_col->lower_bound(4), (opossum::ValueID)2);
-//   EXPECT_EQ(dict_col->upper_bound(4), (opossum::ValueID)3);
-
-//   EXPECT_EQ(dict_col->lower_bound(5), (opossum::ValueID)3);
-//   EXPECT_EQ(dict_col->upper_bound(5), (opossum::ValueID)3);
-
-//   EXPECT_EQ(dict_col->lower_bound(15), opossum::INVALID_VALUE_ID);
-//   EXPECT_EQ(dict_col->upper_bound(15), opossum::INVALID_VALUE_ID);
-// }
-
-// TODO(student): You should add some more tests here (full coverage would be appreciated) and possibly in other files.
+namespace opossum {
+
+// TODO: Perhaps we should create an easily testable DictionarySegment here too
+class StorageDictionarySegmentTest : public ::testing::Test {
+ protected:
+  std::shared_ptr<ValueSegment<int>> vc_int = std::make_shared<ValueSegment<int>>();
+  std::shared_ptr<ValueSegment<std::string>> vc_str = std::make_shared<ValueSegment<std::string>>();
```

Comment:
I find the name "vc" inconvenient, it's a segment, not a column. Can we name it that way?

Reply:
Ahh it's not our code. Nevermind.

Reply:
What about vs_int?